xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/ViewFile.xml

```diff
@@ -263,24 +386,140 @@
     </property>
     <property>
       <code>{{velocity output="false"}}
+#macro(customBox $width $height $escapedFilename)
+  #if ($width &amp;&amp; !$width.contains("%"))
+    #set($width = $width + "px")
+  #end
+  #if ($height &amp;&amp; !$height.contains("%"))
+    #set($height = $height + "px")
+  #end
+  &lt;div class="box" style="width: $width; height: $height; overflow: auto;"&gt;
+
+  {{office reference="attach:$escapedFilename" /}}
+
+  &lt;/div&gt;
+#end
+#macro(customPresentationBox $width $height $escapedFilename)
+  #if ($width &amp;&amp; !$width.contains("%"))
+    #set($width = $width + "px")
+  #end
+  #if ($height &amp;&amp; !$height.contains("%"))
+    #set($height = $height + "px")
+  #end
+  &lt;div style="width: $width; height: $height; overflow: auto;"&gt;
+
+    {{office reference="attach:$escapedFilename" /}}
+
+  &lt;/div&gt;
+#end
+#macro (displayContent $escapedFilename, $modalId)
+  &lt;div class="modal fade" id="$modalId" tabindex="-1" role="dialog"
+      aria-labelledby="${modalId}Label"&gt;
+    &lt;div class="modal-dialog" style="max-height: 90vh; width: 80%;" role="document"&gt;
+      &lt;div class="modal-content"&gt;
+        &lt;div class="modal-header"&gt;
+          &lt;button type="button" class="close" data-dismiss="modal" aria-label="Close"&gt;
+            &lt;span aria-hidden="true"&gt;&amp;times;&lt;/span&gt;&lt;/button&gt;
+          &lt;div id="${modalId}Label" class="modal-title"&gt;
+            $escapetool.xml($services.localization.render('rendering.macro.viewFiles.modal.title'))
+          &lt;/div&gt;
+        &lt;/div&gt;
+        &lt;div class="modal-body" style="max-height: 80vh; overflow: auto"&gt;
+
+        #if ($modalId.contains('pdf'))
```

Comment:
`$escapedFilename.endsWith(".pdf")` would be better instead of `contains`. Since we could have an office filename that contains "pdf".